### PR TITLE
Fix the unit tests validating for required rule type fields

### PR DESCRIPTION
**Pull request by rdimitrov**
# Summary

The following PR fixes the unit tests validating for required rule type fields

## Change Type

***Mark the type of change your PR introduces:***

- [x] Bug fix (resolves an issue without affecting existing features)
- [ ] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [ ] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

***Outline how the changes were tested, including steps to reproduce and any relevant configurations. 
Attach screenshots if helpful.***

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [ ] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [ ] Included tests that validate the fix or feature.
- [ ] Checked that related changes are merged.

### CODE REVIEW

File: pkg/profiles/validator_test.go [hunk 1]
```diff
@@ -277,7 +277,7 @@ func makeProfile(opts ...func(*minderv1.Profile)) *minderv1.Profile {
 
 func loadRawRuleTypeDef() (json.RawMessage, error) {
 	// read rule type from disk and set it up as a fixture
-	f, err := os.Open("../../examples/rules-and-profiles/rule-types/github/branch_protection_allow_force_pushes.yaml")
+	f, err := os.Open("../../examples/rules-and-profiles/rule-types/github/branch_protection_allow_fork_syncing.yaml")
```
Comment (JAORMX) [inline pkg/profiles/validator_test.go:280]: doesn't allowing fork sync also have the same issue? the branch is no longer mandatory.

Comment (rdimitrov) [reply pkg/profiles/validator_test.go:280]: We are not relying on the `branch` field anymore, instead we use the `allow_fork_syncing` required field, i.e. https://github.com/mindersec/minder-rules-and-profiles/blob/5e9722f3472294e3b2cac6ffa115c36acef7aacd/rule-types/github/branch_protection_allow_fork_syncing.yaml#L41

Comment (JAORMX) [reply pkg/profiles/validator_test.go:280]: d'oh! that makes sense and somehow I missed it. Thanks for clarifying!

Comment (rdimitrov) [reply pkg/profiles/validator_test.go:280]: Which is interesting because now it makes me wonder why this test worked before...

**Review by JAORMX — DISMISSED** [review]

**Pull request merged**